src/Reflex/Spider/Internal.hs

```diff
@@ -1537,13 +1532,16 @@ getFanSubscribed k f sub = do
       subscribersRef <- liftIO $ newIORef $ error "getFanSubscribed: subscribersRef not yet initialized"
       occRef <- liftIO $ newIORef parentOcc
       when (isJust parentOcc) $ scheduleClear occRef
+#ifdef DEBUG_NODEIDS
+      nid <- liftIO newNodeId
+#endif
       let subscribed = FanSubscribed
             { fanSubscribedCachedSubscribed = fanSubscribed f
             , fanSubscribedOccurrence = occRef
             , fanSubscribedParent = subscription
             , fanSubscribedSubscribers = subscribersRef
 #ifdef DEBUG_NODEIDS
-            , fanSubscribedNodeId = unsafeNodeId f
+            , fanSubscribedNodeId = nid
```

Comment:
Er... I'm not sure if this is right. the previous code only made the node ID depend on `f`; this makes it depend on more. What's semantically right?

Reply:
That is: do you *intend* the call to `unsafeNodeId` to float out? If so, how far?

Reply:
It seems fine to me, I can't think why having the NodeId in it's own unsafePerformIO would matter - just for convenience and having one less 
#ifdef DEBUG_NODEIDS I am guessing?

Reply:
@saulzar it's just a question of what the node ID is supposed to identify.

Reply:
Ideally I would like a 1-to-1 correspondence between node IDs and the cache IORefs that are being allocated.